draft-irtf-cfrg-hpke.md

```diff
@@ -359,6 +359,11 @@ context. The key schedule inputs are as follows:
 * `pkI` - The initiator's public key (optional; default
   value `zero(Npk)`)
 
+Responders SHOULD validate the initiator-provided encapsulated key for
+correctness. For example, when using a DH-based KEM, the responder should
+check that the resulting key share share is valid, i.e., a point on the
```

Comment:
"key share share"

I am not sure what the "resulting key share" is.
I would instead ask that the recipient/responder should check that pkI and pkE are valid.

Reply:
Good suggestion -- thanks!

Reply:
The sender should then also check if pkR is valid. For secrecy, that's maybe even more important, because it's “before the fact”.